ddtrace/internal/writer.py

```diff
@@ -407,12 +410,22 @@ def _send_payload(self, payload, count):
                 if payload is not None:
                     self._send_payload(payload, count)
         elif response.status >= 400:
-            log.error(
-                "failed to send traces to Datadog Agent at %s: HTTP error status %s, reason %s",
+            msg = "failed to send traces to Datadog Agent at %s: HTTP error status %s, reason %s"
+            log_args = (
                 self.agent_url,
                 response.status,
                 response.reason,
             )
+            # Append the payload if requested
+            if self._log_error_payloads:
+                msg += ", payload %s"
+                # If the payload is bytes then hex encode the value before logging
+                if isinstance(payload, six.binary_type):
+                    log_args += (binascii.hexlify(payload).decode(),)
```

Comment:
this is probably going to be quite expensive with large payloads but i guess this is to be expected